llvm/lib/Analysis/DependenceAnalysis.cpp

```diff
@@ -3569,6 +3569,123 @@ bool DependenceInfo::invalidate(Function &F, const PreservedAnalyses &PA,
          Inv.invalidate<LoopAnalysis>(F, PA);
 }
 
+// Check that memory access offsets in V are multiples of array element size
+// EltSize. Param records the first parametric expression. If the scalar
+// evolution V contains two or more parameters, we check that the subsequent
+// parametric expressions are multiples of the first parametric expression
+// Param.
+static bool checkOffsets(ScalarEvolution *SE, const SCEV *V, const SCEV *&Param,
```

Comment:
Name suggestion: `isKnownMultipleOf`, analogous to e.g. `ScalarEvoluton::isKnownToBeAPowerOfTwo`. Maybe even move it to be a member of ScalarEvolution?